src/notebooks/phdi_analytics.py

```diff
@@ -0,0 +1,297 @@
+    start_date = dt.date(2021, 6, 1)
+    end_date = dt.date(2022, 3, 31)
+
+    points = [generate_point(center, radius) for _ in range(number_of_points)]
+
+    # Generating the synthetic data as a dataframe
+    latitudes = []
+    longitudes = []
+    personids = random.sample(range(len(points)), len(points))
+    first_names = []
+    last_names = []
+    addresses = []
+    fake = Faker()
+    zip_codes = []
+    event_dates = []
+    for p in points:
+        latitudes.append(p[0])
+        longitudes.append(p[1])
+        address = fake.address()
+        street = address.split('\n')[0]
+        first_names.append(names.get_first_name())
+        last_names.append(names.get_last_name())
+        addresses.append(street)
+        zip_codes.append(random.sample(FAIRFAX_COUNTY_ZIPS, 1)[0])
+        event_dates.append(generate_random_date(start_date, end_date))
+
+    df = pd.DataFrame({
+        'pprl_generated_id': personids,
+        'first_name': first_names,
+        'last_name': last_names,
+        'street_address': addresses,
+        'zip_code': zip_codes,
+        'latitude': latitudes,
+        'longitude': longitudes,
+        'event_dt': event_dates,
+    })
+
+    with ThreadPoolExecutor() as tpe:
+        data = list(
+            tqdm(
+                tpe.map(geocode, df[['latitude', 'longitude']].itertuples()),
+                total=len(df)
+            )
+        )
+    data_df = pd.DataFrame.from_records(data)
+
+    df['census_tract'] = data_df['state'] + data_df['county'] + data_df['tract']
+    df['census_block_group'] = data_df['state'] +\
+        data_df['county'] + data_df['tract'] + data_df['block'].str[0]
+    df['census_block'] = data_df['geoid']
+
+    is_fully_vax = random.choices([0, 1], weights=(0.4, 0.6), k=len(points))
+    df['is_fully_vax'] = is_fully_vax
+    df['tested_covid_positive'] = df.apply(
+        lambda row: generate_covid_positive(row), 1)
+    df['breakthrough_infection'] = df['is_fully_vax'] *\
+        df['tested_covid_positive']
+
+    df['event_dt'] = pd.to_datetime(df['event_dt'])
+
+    # Write as Pickle File
+    df.to_pickle(os.path.join(OUTPUTS_DIR, 'synthetic_df.pkl'))
+
+    return df
+
+
+def aggregate_df(df, window='30D', metric_label='Breakthrough Infection'):
+    df.sort_values(by='event_dt', ascending=True, inplace=True)
+
+    if metric_label.lower() == 'breakthrough infection':
+        metric = 'breakthrough_infection'
+
+    if metric_label.lower() == 'covid infection':
+        metric = 'tested_covid_positive'
+
+    df['rolling_sum'] = df.rolling(window, on='event_dt')[metric].sum()
+    df['rolling_count'] = df.rolling(window, on='event_dt')[metric].count()
+    df['rolling_rate'] = df['rolling_sum'] / df['rolling_count']
+
+    return df
+
+
+def plot_time_series_metric(df, window, metric_label):
+    df_agg = aggregate_df(df, window=window, metric_label=metric_label)
+
+    window_length_label = re.findall(r'\d+', window)[0]
```

Comment:
The use of regex here is unnecessary. You should either use `if else` statements to check the values of `window`, or ensure that `window` is an integer and assume that it means number of days. Either way, using regex here gives a big code smell.